### ISSUE TITLE
Take in script arguments via --option for predicate and programmable filter

### ISSUE THREAD

**Issue by hobu** [PDAL/PDAL#931]
It would be convenient to be able to override some script parameters via option so that a script could stay constant.

Something like 

```
        <Filter type="filters.predicate">
            <Option name="module">anything</Option>
            <Option name="function">filter</Option>
            <Option name="variable">keep_classes = [1,2]</Option>
            <Option name="variable">something=somethingelse</Option>
            <Option name="source">
import numpy as np

def filter(ins,outs):
    cls = ins['Classification']

    # keep_classes = [1,2]

    # Use the first test for our base array.
    keep = np.equal(cls, keep_classes[0])

    # For 1:n, test each predicate and join back 
    # to our existing predicate array
    for k in range(1,len(keep_classes)):
        t = np.equal(cls, keep_classes[k])
        keep = keep + t

    outs['Mask'] = keep
    return True
            </Option>
```

All `variable` options should be eval'd in the context of the module and available as a globals. This approach would allow for convenient override via command-line argument:

`pdal pipeline myfile.xml --filters.predicate.variable="keep_classes=[1,2]"`

**Comment by abellgithub** [comment]
@hobu: I haven't kept up with Python integration.  Is this still a thing?

**Comment by hobu** [comment]
Implemented as `gdalargs` option in #1546
